infoblox/resource_nsgroup_delegation.go

```diff
@@ -0,0 +1,163 @@
+package infoblox
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/sky-uk/skyinfoblox"
+	"github.com/sky-uk/skyinfoblox/api/nsgroupdelegation"
+	"github.com/sky-uk/terraform-provider-infoblox/infoblox/util"
+	"net/http"
+)
+
+func resourceNSGroupDelegation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNSGroupDelegationCreate,
+		Read:   resourceNSGroupDelegationRead,
+		Update: resourceNSGroupDelegationUpdate,
+		Delete: resourceNSGroupDelegationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The name of the name server group",
+				Required:    true,
+			},
+			"comment": {
+				Type:         schema.TypeString,
+				Description:  "Comment field",
+				Optional:     true,
+				ValidateFunc: util.CheckLeadingTrailingSpaces,
+			},
+			// Note: Only name and address in the external server struct are used by this resource.
+			"delegate_to": util.ExternalServerListSchema(false, true),
+		},
+	}
+}
+
+func resourceNSGroupDelegationCreate(d *schema.ResourceData, m interface{}) error {
+
+	var nsGroupDelegationObject nsgroupdelegation.NSGroupDelegation
+	client := m.(*skyinfoblox.InfobloxClient)
+
+	if v, ok := d.GetOk("name"); ok && v != "" {
+		nsGroupDelegationObject.Name = v.(string)
+	}
+	if v, ok := d.GetOk("comment"); ok && v != "" {
+		nsGroupDelegationObject.Comment = v.(string)
+	}
+	if v, ok := d.GetOk("delegate_to"); ok && v != nil {
+		servers := []map[string]interface{}{}
+		for _, server := range v.([]interface{}) {
+			servers = append(servers, server.(map[string]interface{}))
+		}
+		nsGroupDelegationObject.DelegateTo = util.BuildExternalServerListFromT(servers)
+	}
+
+	createNSGroupDelegationAPI := nsgroupdelegation.NewCreate(nsGroupDelegationObject)
+	err := client.Do(createNSGroupDelegationAPI)
+	httpStatus := createNSGroupDelegationAPI.StatusCode()
+	if err != nil || httpStatus < http.StatusOK || httpStatus >= http.StatusBadRequest {
+		return fmt.Errorf("Infoblox NS Group Delegation create for %s failed with status code %d and error: %+v", nsGroupDelegationObject.Name, httpStatus, err)
```

Comment:
err will only be defined in case the http call returns an error, in case of a http status 400 for example , err will be empty.